base_layer/core/src/base_node/rpc/query_service.rs

```diff
@@ -280,13 +287,18 @@ impl<B: BlockchainBackend + 'static> Service<B> {
                     .ok_or_else(|| Error::HeaderNotFound {
                         height: current_header.height + 1,
                     })?;
+            if current_header.height == end_height {
+                next_header_to_request = current_header.hash().to_vec();
+                break; // Stop if we reach the end height}
+            }
         }
 
-        let has_next_page = (end_header.height - current_header.height) > 0;
+        let has_next_page = (end_height - current_header.height) > 0;
```

Comment:
This doesn't look right. Higher up you used number of utxos to determine if there was another page